core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -718,6 +718,37 @@ class QualificationAppInfo(
       appId, hasFailures)
   }
 
+  def getAllSQLDurations: Seq[Long] = {
+    sqlIdToInfo.map { case (_, info) =>
+      info.rootExecutionID match {
+        // We return the duration if sqlId doesn't have a rootExecutionID or if the rootExecutionID
+        // is the same as the sqlId. In some cases, the child completes the execution before the
+        // parent, so we need to check if the child's duration is within the parent's duration.
+        // We add the child's duration to the parent's duration if it's not within the parent's
+        // duration.
+        case Some(rootExecutionID) if rootExecutionID == info.sqlID =>
+          info.duration.getOrElse(0L)
+        case Some(rootExecutionID) if rootExecutionID != info.sqlID =>
+          val rootExecutionInfo = sqlIdToInfo.get(rootExecutionID)
+          if (rootExecutionInfo.nonEmpty) {
```

Comment:
Is this a typo? Usually `nonEmpty` is for itertable objects but in this case a value is retrieved from the hashMap which may not be dfined.
This might cause NPE if the SQLevent is processed before the rootSQL has been added to the hashMap.